test/expr/cast/objc_coerce_array.swift

```diff
@@ -7,9 +7,13 @@ var x = 1
 _ = [x] as [NSNumber]
 
 _ = ["x":["y":"z","a":1]] as [String : [String : AnyObject]]
+// expected-warning@-1{{heterogeneous collection literal could only be inferred to '[String : AnyObject]'; add explicit type annotation if this is intentiona}}
```

Comment:
This is another instance of `let _: [Any?] = [1, [2, [3]]]` - we shouldn't diagnose it because there is an explicit coercion here.

Reply:
This one is interesting. Is the inner literal being marked as defaulted by mistake?

Reply:
Would we warn about `let _: [Any] = []`?